src/uproot/source/fsspec.py

```diff
@@ -0,0 +1,139 @@
+import uproot.source.chunk
+
+
+class FSSpecSource(uproot.source.chunk.Source):
+    """
+    Args:
+        file_path (str): A URL for the file to open.
+        **kwargs (dict): any extra arguments to be forwarded to the particular
+            FileSystem instance constructor. This might include S3 access keys,
+            or HTTP headers, etc.
+
+    A :doc:`uproot.source.chunk.Source` that uses FSSpec's cat_ranges feature
+    to get many chunks in one request.
+    """
+
+    def __init__(self, file_path, **kwargs):
+        import fsspec.core
+
+        # Remove uproot-specific options (should be done earlier)
+        # TODO: is timeout always valid?
+        opts = {
+            k: v
+            for k, v in kwargs.items()
+            if k
+            not in {
+                "file_handler",
+                "xrootd_handler",
+                "http_handler",
+                "s3_handler",
+                "object_handler",
+                "max_num_elements",
+                "num_workers",
+                "num_fallback_workers",
+                "use_threads",
+                "begin_chunk_size",
+                "minimal_ttree_metadata",
```

Comment:
I think this is the best place to remove the uproot-specific options (addressing comment in code above).

The CI was failling because uproot-specific options were being passed into which should have been removed due to the list of options being updated after this was written.

I guess it would be a good idea to avoid this hardcoded list of uproot options. I don't think it's easy to do as of right now. Looks like the list of valid options is maintained on `reading.py`, if it doesn't exist already perhaps we could add a validation of options (enforce user didn't use an invalid keyword) from a list of valid options and import this list to remove uproot-specific options?

Reply:
Update: load options dictionary from `uproot.reading.open.defaults`